crates/worker/src/partition_processor_manager.rs

```diff
@@ -311,7 +425,89 @@ impl PartitionProcessorManager {
         }
     }
 
-    pub async fn apply_plan(&mut self, actions: &[Action]) -> anyhow::Result<()> {
+    async fn on_control_processors(
+        &mut self,
+        control_processor: MessageEnvelope<ControlProcessors>,
+    ) -> Result<(), Error> {
+        let (_, control_processors) = control_processor.split();
+
+        let partition_table = self
+            .metadata
+            .wait_for_partition_table(control_processors.min_partition_table_version)
+            .await?;
+
+        for control_processor in control_processors.commands {
+            self.on_control_processor(control_processor, &partition_table)
+                .await?;
+        }
```

Comment:
I'm a little concerned about critical control loops with inner awaits. Best if we can execute concurrently but it might be too much of a hassle given how everything is structured. Alternatives could be to do fail early on potential blocking points, or add a timeout to limit waiting time upper bound.

Reply:
Yes, this is not good. As a temporary band-aid, I will replace the `sends` with `try_send` as you've suggested. It still does not make control loop completely await-free. I will take a look at this in a separate step.